csrc/multidevice/communication.cpp

```diff
@@ -334,16 +334,23 @@ c10::intrusive_ptr<c10d::Work> postAllgather(
     c10d::Backend* backend,
     at::Tensor input_tensor,
     at::Tensor output_tensor) {
+  
+  // input and output tensors maybe strided (tensor with shape [m, n, k] and strides [1, k*m, m]), so we flatten them to match the ProcessGroupNCCL contiguity requirements.
+  // Presegmentation pass `makeReshardingContiguous` ensures that the tvs are contiguous
+  // and HostIrExecutor validates the tensor against the tv allocation domain.
+
+  auto flattened_output_tensor = output_tensor.as_strided({output_tensor.numel()}, {1});
```

Comment:
Also check contiguity of communication->in() and out()?

Reply:
It is already enforced by `makeReshardingContiguous` pass so I am not duplicating it here.

Reply:
Nit: makeReshardingContiguous is a bit too far and many changes could happen in between. For example, makeReshardingContiguous runs before segmentation and postSingleCommunication is at runtime. makeReshardingContiguous works on fusion IR e.g. set and reduce, and postSingleCommunication works on host IR e.g. Communication.

Reply:
I am running into some test failures with a contiguity check here for the manual tests in `test_multidevice_host_ir.cpp`. Since these tests do not set an allocation domain, we have the contiguity set to false. How cumbersome is it to require manual tests also to have the allocation domain set correctly?
CC: @samnordmann

Reply:
> How cumbersome is it to require manual tests also to have the allocation domain set correctly?

IIUC, it should be set correctly, so let's set it correctly. The change can't be too large because test_multidevice_host_ir.cpp is <500 lines and that file has <20 calls to `make*Tensor`, many of which are `Contig` already.

Reply:
hey Priya, I am not sure how cumbersome that is -- but if needed feel free to do it, and please let me know how it looks like.
Let me know also if you need help